src/www/app/app.ts

```diff
@@ -591,7 +591,7 @@ export class App {
 
   private registerUrlInterceptionListener(urlInterceptor: UrlInterceptor) {
     urlInterceptor.registerListener(url => {
-      if (!url || !unwrapInvite(url).startsWith('ss://')) {
+      if (!url || !(unwrapInvite(url).startsWith('ss://') || unwrapInvite(url).startsWith('ssconf://'))) {
```

Comment:
We need to simplify and clarify how we handle the urls. This is not sustainable and is why we are running into these issues.

First, define a `IsOutlineAccessKey(url)` function. It must have tests. It can be in app.spec.ts or some other library you create. Use that everywhere you need to check if a URL is for Outline, including here, but also:
https://github.com/Jigsaw-Code/outline-client/blob/0851dc70bc4bfcb216702cf3cc6a331271f44d92/src/www/app/app.ts#L46

For the URL extraction in https://github.com/Jigsaw-Code/outline-client/blob/0851dc70bc4bfcb216702cf3cc6a331271f44d92/src/www/app/app.ts#L44, define some `OUTLINE_URL_SCHEMAS` that you can iterate over. We should consider later a URL extraction from any text, in case someone copied a text message with text+URL. It could leverage the same mechanism.

While looking at this code, I realized this check will break checks of invites with names in the hash:
https://github.com/Jigsaw-Code/outline-client/blob/0851dc70bc4bfcb216702cf3cc6a331271f44d92/src/www/app/app.ts#L36

Please fix that and add the appropriate tests to https://github.com/Jigsaw-Code/outline-client/blob/master/src/www/app/app.spec.ts.

Reply:
I believe my suggestions are straightforward, and they fix bugs that your change is not fixing. I already did the work of finding the places to fix for you.

At a minimum, the two lines I mention above must be fixed. That will be cleaner if you define `IsOutlineAccessKey()`. The tests should be straightforward too, we already have them set up.

We also need to make sure the fixed unwrap is not breaking names for dynamic keys.

Reply:
We discussed offline. The invite page gives us an "Invalid invite" error if used with `ssconf` so we don't need to support it in the client. Let's add a comment to the unwrap method explaining why we don't need to support other schemes.

The hash issue was a misunderstanding of mine. The hash in the invite is url encoded, so it doesn't trigger the test.

So there's no behavior to fix. The IsOutlineAccessKey is a nice to have that will improve readability.